flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphRestartTest.java

```diff
@@ -155,21 +153,12 @@ public void testRestartAutomatically() throws Exception {
 	public void testCancelWhileRestarting() throws Exception {
 		// We want to manually control the restart and delay
 		RestartStrategy restartStrategy = new InfiniteDelayRestartStrategy();
-		Tuple2<ExecutionGraph, Instance> executionGraphInstanceTuple = createExecutionGraph(restartStrategy);
-		ExecutionGraph executionGraph = executionGraphInstanceTuple.f0;
-		Instance instance = executionGraphInstanceTuple.f1;
 
-		// Kill the instance and wait for the job to restart
-		instance.markDead();
-
-		Deadline deadline = TestingUtils.TESTING_DURATION().fromNow();
-
-		while (deadline.hasTimeLeft() &&
-				executionGraph.getState() != JobStatus.RESTARTING) {
-
-			Thread.sleep(100);
-		}
+		SlotProvider slotProvider = new TestingSlotProvider(ignore -> new CompletableFuture<>());
+		ExecutionGraph executionGraph = createSimpleExecutionGraph(restartStrategy, slotProvider);
+		assertEquals(JobStatus.CREATED, executionGraph.getState());
 
+		executionGraph.scheduleForExecution();
 		assertEquals(JobStatus.RESTARTING, executionGraph.getState());
```

Comment:
why is the EG in a restarting state? Previously this was caused by killing the taskmanager, but what is causing this now?

Reply:
This is because the SlotProvider returns non-completed future. And eg is configured as not allow queued scheduling. Thus when scheduleForExecutor, it causes `IllegalArgumentException("The slot allocation future has not been completed yet.")` and then failing and then restarting.

But it is tricky. Maybe I should set allow queued scheduling and fail a future to emulate a real failure.

Reply:
If we fail the future right away the scheduling mode should be irrelevant, correct? It should fail in any case and restart the job.

Reply:
Sure.